pkg/network/protocols/http/ebpf_javatls.go

```diff
@@ -78,7 +106,40 @@ func (p *JavaTLSProgram) GetAllUndefinedProbes() (probeList []manager.ProbeIdent
 	return
 }
 
+// isAttachmentAllowed will return true if the pid must not be attach
+// The filter is based on the process command line matching javaAgentAllowRegex and javaAgentBlockRegex regex
+// javaAgentAllowRegex have an higher priority
+func isAttachmentAllowed(pid uint32) bool {
+	// filter is disabled (default configuration)
+	if javaAgentAllowRegex == nil && javaAgentBlockRegex == nil {
+		return true
+	}
+
+	procCmdline := fmt.Sprintf("%s/%d/cmdline", util.HostProc(), pid)
+	cmd, err := os.ReadFile(procCmdline)
+	if err != nil {
+		log.Debugf("injectionFilter can't open comandline %s : %s", procCmdline, err)
+		return false
+	}
+	fullCmdline := strings.ReplaceAll(string(cmd), "\000", " ") // /proc/pid/cmdline format : arguments are separated by '\0'
+
+	// Allow have an higher priority
+	if javaAgentAllowRegex != nil && javaAgentAllowRegex.MatchString(fullCmdline) {
+		return true
+	}
+	if javaAgentBlockRegex != nil && javaAgentBlockRegex.MatchString(fullCmdline) {
+		return false
+	}
+	// by default we attach the java process
+	return true
```

Comment:
If we made it to here, it mean that at least one of  `javaAgentAllowRegex` or `javaAgentBlockRegex`  is not nil, but we didn't match any of them, so we should deny




```suggestion
	// by default we attach the java process
	return false
```

Reply:
I'm not agree (but it's typical default behavior of allow/block list discussion), as customer 
 * I would like to not attach elasticsearch
 * I would like to attach elasticsearch with an option / version2 / cluster=abc
 
If we deny be default, the customer expect to attach any other process
if he would like to deny by default, feel free to set  blockregex = ".*"

Reply:
talked about that by heart